lib/base/tlsutility.cpp

```diff
@@ -235,16 +257,26 @@ void SetCipherListToSSLContext(const Shared<boost::asio::ssl::context>::Ptr& con
  */
 void SetTlsProtocolminToSSLContext(const Shared<boost::asio::ssl::context>::Ptr& context, const String& tlsProtocolmin)
 {
-	// tlsProtocolmin has no effect since we enforce TLS 1.2 since 2.11.
-	/*
-	std::shared_ptr<SSL_CTX> sslContext = std::shared_ptr<SSL_CTX>(context->native_handle());
-
-	long flags = SSL_CTX_get_options(sslContext.get());
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
```

Comment:
... if this condition doesn’t fire?

Reply:
Then OpenSSL does not support `SSL_CTX_set_min_proto_version()` (was added in 1.1.0). So we end up in a situation where TLS 1.2 is the only protocol version we want to support.

Everything lower than TLS 1.2 is already disabled here: https://github.com/Icinga/icinga2/blob/8c2706dd3647047c43e6804560bbabfbbd8774f5/lib/base/tlsutility.cpp#L78-L85